lab_1/main.py

```diff
@@ -98,7 +213,33 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
+    if not isinstance(unknown_profile, dict):
+        return None
+    if not isinstance(profiles, list):
+        return None
+    if not isinstance(languages, list):
+        return None
+    if not isinstance(top_n, int):
+        return None
+
+    actual_profiles = []
+    if len(languages) == 0:
+        actual_profiles = profiles
+    for i in profiles:
+        if i["name"] in languages:
+            actual_profiles.append(i)
+
+    results = []
+    for profile in actual_profiles:
+        results.append(compare_profiles_advanced(unknown_profile, profile, top_n))
+
+    results = sorted(results, key=lambda x: x["name"])
+    results = sorted(results, key=lambda x: x["score"], reverse=True)
```

Comment:
can you explain what happens here and why you need that?

Reply:
First - dictionaries in list "results" are sorted by names of languages: it means after sorting by score (the next line) dictionaries with the same score will be arranged in alphabetical order.